File: Easy/168. Excel Sheet Column Title/168. Excel Sheet Column Title.js
/**
 * @param {number} columnNumber
 * @return {string}
 */
var convertToTitle = function (columnNumber) {
    let result = '';

    while (columnNumber > 0) {
        columnNumber--; // Adjust to 0-based index
        const remainder = columnNumber % 26;
        const digit = String.fromCharCode('A'.charCodeAt(0) + remainder);
        result = digit + result;
        columnNumber = Math.floor(columnNumber / 26);
    }

    return result;
};

File: Easy/168. Excel Sheet Column Title/168. Excel Sheet Column Title.py
class Solution:
    def convertToTitle(self, columnNumber: int) -> str:
        result = ""

        while columnNumber > 0:
            columnNumber -= 1  # Adjust to 0-based index
            remainder = columnNumber % 26
            digit = chr(ord("A") + remainder)
            result = digit + result
            columnNumber //= 26

        return result
